# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/kde/koffice/hotfix-binutils-gcc.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Hotfix needed for new binutils (bug is in gcc)
(see http://gcc.gnu.org/bugzilla/show_bug.cgi?id=16625)

--- ./kchart/kdchart/KDChartEnums.h.orig	2004-08-13 14:23:43.000000000 +0200
+++ ./kchart/kdchart/KDChartEnums.h	2004-08-13 14:24:41.000000000 +0200
@@ -279,22 +279,25 @@
                                 return "TopCenter";
                             case PosTopRight:
                                 return "TopRight";
+			}
+                        switch( type ) {
                             case PosCenterLeft:
                                 return "CenterLeft";
                             case PosCenter:
                                 return "Center";
                             case PosCenterRight:
                                 return "CenterRight";
+			}
+                        switch( type ) {
                             case PosBottomLeft:
                                 return "BottomLeft";
                             case PosBottomCenter:
                                 return "BottomCenter";
                             case PosBottomRight:
                                 return "BottomRight";
-                            default: // should not happen
-                                qDebug( "Unknown content position" );
-                                return "TopLeft";
                         }
+                        qDebug( "Unknown content position" );
+                        return "TopLeft";
                     }
 
 
